R/github.R

```diff
@@ -27,10 +28,24 @@ github_commit <- function(username, repo, ref = "master") {
 #'
 #' @keywords internal
 #' @noRd
-github_pat <- function() {
+github_pat <- function(quiet = TRUE) {
   pat <- Sys.getenv('GITHUB_PAT')
   if (identical(pat, "")) return(NULL)
 
-  message("Using github PAT from envvar GITHUB_PAT")
+  if (!quiet) {
+    message("Using github PAT from envvar GITHUB_PAT")
+  }
   pat
 }
+
+github_DESCRIPTION <- function(username, repo, subdir = NULL, ref = "master", host = "api.github.com", ...) {
+
+  url <- file.path(paste0("https://", host),
+                   "repos", username, repo, "contents", paste0(subdir, "DESCRIPTION"))
+  url <- paste0(url, "?=", URLencode(ref))
+
+  tmp <- tempfile()
+  download(tmp, url, auth_token = github_pat())
+
+  rawToChar(base64enc::base64decode(gsub("\\\\n", "", fromJSONFile(tmp)$content)))
```

Comment:
This is the current blocker for this, we need to decode the base64 encoded content, so I guess would have to write a R only version of `base64decode()`.

Reply:
Update: now have this